test/sdp/media/ersip_sdp_ice_candidate_test.erl

```diff
@@ -0,0 +1,98 @@
+    ?assertEqual(45664,                                                 ersip_sdp_ice_candidate:port(IceCandidate2)),
+    ?assertEqual(<<"srflx">>,                                           ersip_sdp_ice_candidate:cand_type(IceCandidate2)),
+    ?assertEqual({ip6, {65152, 0, 0, 0, 26230, 47871, 65180, 61002}},   ersip_sdp_ice_candidate:rel_addr(IceCandidate2)),
+    ?assertEqual(8998,                                                  ersip_sdp_ice_candidate:rel_port(IceCandidate2)),
+    ?assertEqual([{<<"generation">>, <<"0">>}, {<<"time">>, <<"12">>}], ersip_sdp_ice_candidate:attrs(IceCandidate2)),
+
+    {ok, IceCandidate3} = parse(<<"1467250027 1 udp 2122260223 192.168.0.196 46243 typ host generation 0">>),
+    ?assertEqual(<<"1467250027">>,              ersip_sdp_ice_candidate:foundation(IceCandidate3)),
+    ?assertEqual(1,                             ersip_sdp_ice_candidate:component_id(IceCandidate3)),
+    ?assertEqual(<<"udp">>,                     ersip_sdp_ice_candidate:transport(IceCandidate3)),
+    ?assertEqual(2122260223,                    ersip_sdp_ice_candidate:priority(IceCandidate3)),
+    ?assertEqual({ip4, {192, 168, 0, 196}},     ersip_sdp_ice_candidate:connection_address(IceCandidate3)),
+    ?assertEqual(46243,                         ersip_sdp_ice_candidate:port(IceCandidate3)),
+    ?assertEqual(<<"host">>,                    ersip_sdp_ice_candidate:cand_type(IceCandidate3)),
+    ?assertEqual(undefined,                     ersip_sdp_ice_candidate:rel_addr(IceCandidate3)),
+    ?assertEqual(undefined,                     ersip_sdp_ice_candidate:rel_port(IceCandidate3)),
+    ?assertEqual([{<<"generation">>, <<"0">>}], ersip_sdp_ice_candidate:attrs(IceCandidate3)),
+
+    ok.
+
+parse_error_test() ->
+    ?assertMatch({error, {invalid_ice_candidate, _}}, parse(<<"2 1 UDP 1694498815 2001:420:c0e0:1005::61">>)),
+    ?assertMatch({error, {invalid_ice_candidate, _}}, parse(<<"2 1 UDP 1694498815 2001:420:c0e0:1005::61 45664 typ srflx raddr fe80::6676:baff:fe9c:ee4a key_no_value">>)),
+    ?assertMatch({error, {invalid_ice_candidate, _}}, parse(<<"2 1 UDP 0 2001:420:c0e0:1005::61 45664 typ srflx">>)),
+
+    ok.
+
+assemble_test() ->
+    IceCandidateBin = <<"2 1 UDP 1694498815 2001:420:c0e0:1005::61 45664 typ srflx raddr fe80::6676:baff:fe9c:ee4a rport 8998 generation 0 time 12">>,
+    ?assertEqual(IceCandidateBin, reassemble(IceCandidateBin)),
+
+    ok.
+
+%%%===================================================================
+%%% Helpers
+%%%===================================================================
+
+parse(Bin) ->
+    case ersip_sdp_ice_candidate:parse(Bin) of
+        {ok, IceCandidate, <<>>} ->
+            {ok, IceCandidate};
+        {ok, IceCandidate, <<?crlf>>} ->
+            {ok, IceCandidate};
+        {error, _} = Error ->
+            Error
+    end.
+
+reassemble(Bin) ->
+    {ok, IceCandidate} = parse(Bin),
+    iolist_to_binary(ersip_sdp_ice_candidate:assemble(IceCandidate)).
```

Comment:
I'm not sure, should I add the ersip_sdp_ice_candidate:assemble_bin/1 as it has done in ersip_sdp or not?

Reply:
Yes, please add assemble_bin/1

Reply:
done